airsonic-main/src/main/java/org/airsonic/player/service/jukebox/AudioPlayer.java

```diff
@@ -49,16 +52,68 @@
 
     private final InputStream in;
     private final Listener listener;
-    private final SourceDataLine line;
     private final AtomicReference<State> state = new AtomicReference<State>(PAUSED);
+    private SourceDataLine line;
+    private AudioFormat format;
+    private DataLine.Info info;
     private FloatControl gainControl;
 
-    public AudioPlayer(InputStream in, Listener listener) throws Exception {
+    public AudioPlayer(InputStream in, String command, Listener listener) throws Exception {
         this.in = in;
         this.listener = listener;
 
-        AudioFormat format = new AudioFormat(AudioFormat.Encoding.PCM_SIGNED, 44100.0F, 16, 2, 4, 44100.0F, true);
-        line = AudioSystem.getSourceDataLine(format);
+        // SampleRate
+        String sampleRate = "";
+        Pattern r1 = Pattern.compile("\\-ar (.*?)\\ ");
```

Comment:
please compile these patterns outside the method so they're not compiled again and again?